docs/ecosystem/oracles/pragma.md

```diff
@@ -1,3 +1,118 @@
 # Pragma Oracle
 
-Under construction 🏗️
+Pragma is the decentralized, transparent and composable oracle network, leveraging state-of-the-art zero-knowledge cryptography. We partner with the biggest market makers and the most liquid exchanges who sign and timestamp their own high quality, robust data and send it directly on-chain.
+
+Our data feeds are live on 🥕 Kakarot, where they leverage the power of the ZK-EVM. Our code has been audited by leading security researchers at Nethermind and we have more audits coming up soon.
+
+We are currently deployed on the following addresses:
+
+| Network | Cairo Address                                                     | PragmaCaller Solidity Interface                                                                                                  |
+| ------- | ----------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------- |
+| Sepolia | 0x3a99b4b9f711002f1976b3973f4b2031fe6056518615ff0f4e6dd829f972764 | [0x7491cA3699701a187C1a17308338Ad0bA258B082](https://sepolia.kakarotscan.org/address/0x7491cA3699701a187C1a17308338Ad0bA258B082) |
+| Mainnet | **⏳ Soon**                                                        | **⏳ Soon**                                                                                                                       |
+
+# Consuming our Data Feeds
+
+You can get started with Pragma in just a few minutes. This guide will walk you through the process of consuming data from Pragma's oracle network, either from a Solidity or Cairo contract.
+
+You can find the list of supported assets in our main documentation [here](https://docs.pragma.build/Resources/Cairo%201/data-feeds/supported-assets).
+
+## Cairo
```

Comment:
I would remove this from our docs as they're EVM-focused. wdyt @Eikix ?

Reply:
Agreed, unless you have specific Kakarot data feeds in your docs. Maybe if you have a tag in your docs to indicate that a specific feed is supported on Kakarot or if we support 100% of your data feeds?

Reply:
yes im talking about the cairo part.
@akhercha Considering that your architecture is entirely different would this not just add extra complexity?

Reply:
AFAIK it should be fine? We're just gonna build wrappers (one per pair/feed) around the `PragmaCaller` with the `ChainlinkAggregatorV3` interface

Reply:
yep, i'm just wondering whether it makes sense to do this because in the end it all ends up being redirected to the same contract, the only thing you simplify is the ascii encoding of the pair

Reply:
it's only to make it easy for chainlink consumers to migrate to pragma as a first step, even if it doesnt make that much sense it will make their life easier initially while porting to kkrt.